### PR TITLE
DEPR: make arguments keyword only in to_clipboard

### PR DESCRIPTION
**Pull request by rsm-23**
- [x] xref #54229 
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added an entry in the latest `doc/source/whatsnew/v2.2.0.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by rsm-23** [comment]
@mroeschke is anything more required here?

File: pandas/core/generic.py [hunk 1]
```diff
@@ -3099,7 +3099,7 @@ def to_pickle(
 
     @final
     def to_clipboard(
-        self, excel: bool_t = True, sep: str | None = None, **kwargs
+        self, *, excel: bool_t = True, sep: str | None = None, **kwargs
```
Comment (mroeschke) [inline pandas/core/generic.py:3102]: This would still need a deprecation

File: pandas/core/generic.py [hunk 2]
```diff
@@ -3098,8 +3098,11 @@ def to_pickle(
         )
 
     @final
+    @deprecate_nonkeyword_arguments(
+        version="3.0", allowed_args=["self"], name="to_clipboard"
+    )
     def to_clipboard(
-        self, excel: bool_t = True, sep: str | None = None, **kwargs
+        self, *, excel: bool_t = True, sep: str | None = None, **kwargs
```
Comment (mroeschke) [inline pandas/core/generic.py:3105]:
```suggestion
        self, excel: bool_t = True, sep: str | None = None, **kwargs
```

**Comment by rsm-23** [comment]
@mroeschke added test for the same.

**Comment by mroeschke** [comment]
Just a merge conflict that needs fixing

**Comment by rsm-23** [comment]
@mroeschke resolved

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @rsm-23

**Pull request merged**